python/rikai/mixin.py

```diff
@@ -90,6 +90,51 @@ def __matmul__(self, style: Union[dict, "rikai.viz.Style"]) -> Drawable:
         return style(self)
 
 
+class ComposableDrawable:
+    """Mixin for a drawable that can be composed with others"""
+
+    @abstractmethod
+    def left_top(self):
+        """return a tuple of the left top axis (xmin, ymin)"""
+
+    @abstractmethod
+    def aligned(
+        self, base_drawable: ComposableDrawable, algorithm: str = "left-top"
+    ):
+        """return a new instance that is aligned with the base drawable,
```

Comment:
Lets follow with the numpy docstring style here?

https://numpydoc.readthedocs.io/en/latest/format.html